pkg/targetgroupbinding/resource_manager.go

```diff
@@ -405,14 +400,73 @@ func (m *defaultResourceManager) registerPodEndpoints(ctx context.Context, tgARN
 		if err != nil {
 			return err
 		}
-		if !networking.IsIPWithinCIDRs(podIP, vpcCIDRs) {
+
+		if !networking.IsIPWithinCIDRs(podIP, albCIDRs) {
+			m.logger.Info(fmt.Sprintf("On %s, the podIP %s does not belong to any of it's LB's CIDRs %v, hence we are setting AvailabilityZone=all", tgARN, podIP, albCIDRs))
 			target.AvailabilityZone = awssdk.String("all")
 		}
 		sdkTargets = append(sdkTargets, target)
 	}
 	return m.targetsManager.RegisterTargets(ctx, tgARN, sdkTargets)
 }
 
+func (m *defaultResourceManager) getAlbVpcCidr(ctx context.Context, tgARN string) ([]netip.Prefix, error) {
```

Comment:
Thanks for the contribution. My concern is for every reconciliation on IP target group, this PR adds 2 ELBv2 API calls, `DescribeTargetGroupsAsList` and `DescribeLoadBalancersAsList`. If the user has a large scale of resources, this may significantly increase the API calls or controller reconciliation latency during the situation like controller restart/auto-reconcile.